### ISSUE TITLE
SyntaxError: 'import' and 'export' may appear only with 'sourceType: "module"' (8:0)

### ISSUE THREAD

**Issue by Mrna1** [stryker-mutator/stryker#1229]
Hi all
I am having troubles with babel when I try to run stryker run.
It works fine when I compile my code.

Stack trace: 14:14:11 (23990) INFO ConfigReader Using stryker.conf.js in the current working directory.
14:14:12 (23990) INFO TypescriptConfigEditor Loading tsconfig file /home/frantisek/Perforce/fmanak_workspace/Platform_Solutions/Brno/GLMatching_Mainline/tsconfig.json
14:14:12 (23990) INFO InputFileResolver Found 140 of 17285 file(s) to be mutated.
14:14:12 (23990) INFO InitialTestExecutor Starting initial test run. This may take a while.
14:14:13 (23990) ERROR StrykerCli an error occurred { Error: Could not instrument "/home/frantisek/.../src/account/index.ts" for code coverage. Inner error: SyntaxError: BABEL_PARSER_SOURCETYPE_MODULE_REQUIRED (undefined) SyntaxError: 'import' and 'export' may appear only with 'sourceType: "module"' (8:0)
    at _class.raise (/home/frantisek/.../node_modules/@babel/parser/lib/index.js:3939:15)

Could you help me with that? According to https://github.com/babel/babelify/issues/157
it is inconsistent of babel version.

**Comment by simondel** [comment]
@Mrna1 Thanks for opening this issue! Could you please give us your current `stryker.conf.js` file, `.bablrc` file and `npm test` command?

**Comment by Mrna1** [comment]
Hi, here they are.

stryker.conf.js:
```
module.exports = function(config) {
  config.set({
    mutator: "typescript",
    packageManager: "npm",
    reporters: ["clear-text", "progress"],
    testRunner: "mocha",
    transpilers: [],
    testFramework: "mocha",
    coverageAnalysis: "perTest",
    tsconfigFile: "tsconfig.json",
    mutate: ["src/**/*.ts"]
  });
};
```
.babelrc.js

```
const fs = require("fs");
const path = require("path");
const { compilerOptions } = require("./tsconfig");

const sourceDirectory = "src";

const outputDir = path.resolve(__dirname, compilerOptions.outDir);
const directory = path.resolve(__dirname, sourceDirectory);

const modules = fs.readdirSync(directory)
  .filter((file) => fs.statSync(path.resolve(directory, file)).isDirectory())
  .join('|');

const alias = [
  [
    new RegExp(`^@(${modules})$`),
    (groups) => path.resolve(outputDir, `${sourceDirectory}/${groups[0]}/index`)
  ],
  [
    /^@mocks\/(.+)$/,
    (groups) => path.resolve(outputDir, `${"test"}/mocks/${groups[0]}`)
  ],
];

module.exports = {
  presets: [
    ["@babel/preset-env", {
      targets: {
        browsers: ["ie < 8"],
      },
      modules: "amd"
    }]
  ],
  plugins: [
    ["babel-plugin-preserve-comment-header", {
      "pattern": "@TestVersion"
    }],
    ["module-resolver", {
      extensions: [".js", ".ts"],
      resolvePath(sourcePath, currentFile) {
        const found = alias.find((row) => row[0].test(sourcePath));
        if (found) {
          const result = found[1](found[0].exec(sourcePath).slice(1));
          const current = path.dirname(path.relative(__dirname, currentFile));
          return path.relative(current, result).split(path.sep).join('/');
        }
        return sourcePath;
      }
    }],
  ],
};

```

npm test command:
I do not have any extend command just use Environment variables:
`TS_NODE_COMPILER_OPTIONS={"module": "commonjs", "moduleResolution": "node", "target": "es5"}
`

**Comment by simondel** [comment]
Thanks for the info! Right now we don't support babel transpilation with TypeScript. To get Stryker to run anyways you should be able to set `transpilers: ['typescript'],` in your Stryker config.

Could you let me know if that works for you? I'll create a new issue for TypeScript support with Babel (7)

**Comment by Mrna1** [comment]
Thank you it helped, just failed with another issue with bin.js.map, but I will investigate it later.

**Comment by simondel** [comment]
I'm closing this issue for now. We have a separate issue to track TypeScript support with Babel. We do have Babel 7 support, so this may already work.

**Comment by stephen-smith** [comment]
I'm getting this, now.

I have a small vue+ts project: https://gitlab.com/bss03/vue-webpack-ts-lambda

That's the commit before I started trying to add stryker, it's finally reporting 100% coverage on at least one file, though I know the tests are not exhaustive in the least.

I followed the stryker guide: https://github.com/stryker-mutator/stryker-handbook/blob/master/stryker/guides/vuejs.md#vuejs

I got the failure with the "jest" testRunner, but that's not how I run my tests anyway, so I switched to using the "command" testRunner with the command being `npm run test:unit` and got the same error.  So, I actually think the failure is before the testRunner is involved in the initial test run.

Logs on pastebin: https://pastebin.com/WWdPEXf4

Any help would be appreciated.
